dashdotdb/controllers/tokens.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
+from uuid import uuid4
+
+from dashdotdb.models.dashdotdb import db
+from dashdotdb.models.dashdotdb import Token
+from dashdotdb.models.dashdotdb import LatestTokens
+
+
+def close_token(uuid, data_type):
+    db_token = db.session.query(Token) \
+        .filter(Token.uuid == uuid, Token.data_type == data_type).first()
+    if db_token is None:
+        return 'token not found', 404
+    with db.session.begin():
+        db_token.is_open = False
+        latest_token = db.session.query(LatestTokens) \
+            .filter(Token.data_type == data_type,
+                    Token.id == LatestTokens.token_id).first()
+        if latest_token is None:
+            db.session.add(LatestTokens(token_id=db_token.id))
```

Comment:
Should we accumulate all the latest tokens or have ONE latest token in the DB at any given moment of time for each data type?

Reply:
The tokens table will still keep all the tokens, but the latesttokens table will only ever point to the last closed token for a data type.  What would be the use case to have more than 1 latesttoken for each type?

If I understand the original implementation correctly, we always assume the most recent token has all the most recent and relevant data.  I kept that assumption with this approach.

Reply:
Yes, you're right. I was under the impression that, looking at the LatestTokens table, I would see many many tokens there, not only the latest one, but I might have overlooked it :)